crates/ruff/src/rules/pycodestyle/rules/lambda_assignment.rs

```diff
@@ -111,14 +121,111 @@ pub fn lambda_assignment(checker: &mut Checker, target: &Expr, value: &Expr, stm
     }
 }
 
-fn function(name: &str, args: &Arguments, body: &Expr, stylist: &Stylist) -> String {
+/// Extract the argument types and return type from a `Callable` annotation.
+/// The `Callable` import can be from either `collections.abc` or `typing`.
+/// If an ellipsis is used for the argument types, an empty list is returned.
+/// The returned values are cloned, so they can be used as-is.
+fn extract_types(ctx: &Context, annotation: &Expr) -> Option<(Vec<Expr>, Expr)> {
```

Comment:
Is there any way to write unit tests for this function? This is mainly regarding the `Context`. If there are any examples on how to create a mock `Context`, I could do that. If not, should we provide some functionality to do that to facilitate writing unit tests? (`Context` is being used in a lot of helper functions)

Reply:
Sadly no, but I would definitely be interested in mechanisms for creating `Context` in tests.

Mocking would be one approach. Another would be to provide some primer Python code that we could parse and traverse, then use the `Context` that results from traversing that code, and pass it to (e.g.) this function. That might be hard to do though, since `Context` is really driven by `Checker` and hard to decouple...